set-based-validation/src/main/java/io/axoniq/dev/samples/interceptor/AccountCreationDispatchInterceptor.java

```diff
@@ -0,0 +1,33 @@
+package io.axoniq.dev.samples.interceptor;
+
+import java.util.List;
+import java.util.function.BiFunction;
+
+import org.axonframework.commandhandling.CommandMessage;
+import org.axonframework.messaging.MessageDispatchInterceptor;
+import org.springframework.stereotype.Component;
+
+import io.axoniq.dev.samples.api.CreateAccountCommand;
+import io.axoniq.dev.samples.persistence.EmailRepository;
+
+@Component
```

Comment:
Just a thought, but it might be nice to mark which solution to sharing the set validation this is by adding a little javadoc.
Same would thus apply to the other solutions we have.